crates/optimism/rpc/src/eth/pending_block.rs

```diff
@@ -39,6 +46,53 @@ where
         self.inner.evm_config()
     }
 
+    /// Returns the locally built pending block
+    async fn local_pending_block(&self) -> Result<Option<SealedBlockWithSenders>, Self::Error> {
+        let pending = self.pending_block_env_and_cfg()?;
+        if pending.origin.is_actual_pending() {
+            return Ok(pending.origin.into_actual_pending())
+        }
+
+        let mut lock = self.pending_block().lock().await;
+
+        let now = Instant::now();
+
+        // check if the block is still good
+        if let Some(pending_block) = lock.as_ref() {
+            // this is guaranteed to be the `latest` header
+            if pending.block_env.number.to::<u64>() == pending_block.block.number &&
+                pending.origin.header().hash() == pending_block.block.parent_hash &&
+                now <= pending_block.expires_at
+            {
+                return Ok(Some(pending_block.block.clone()))
+            }
+        }
```

Comment:
I think we can simplify this and always return the latest block, see referenced op-geth code

Reply:
right, there will never be a pending block because it's built centrally by the op labs sequencer? so there will always only be latest block and we can remove this code - right?

Reply:
yes exactly